libs/wingc/src/jsify.rs

```diff
@@ -227,10 +227,71 @@ impl<'a> JSifier<'a> {
 		}
 	}
 
-	fn jsify_type(&self, typ: &TypeAnnotationKind) -> String {
+	fn jsify_type(&self, typ: &TypeAnnotationKind) -> Option<String> {
 		match typ {
-			TypeAnnotationKind::UserDefined(t) => self.jsify_user_defined_type(&t),
-			_ => todo!(),
+			TypeAnnotationKind::UserDefined(t) => Some(self.jsify_user_defined_type(&t)),
+			TypeAnnotationKind::String => Some("string".to_string()),
+			TypeAnnotationKind::Number => Some("number".to_string()),
+			TypeAnnotationKind::Bool => Some("boolean".to_string()),
+			TypeAnnotationKind::Array(t) => {
+				if let Some(inner) = self.jsify_type(&t.kind) {
+					Some(format!("{}[]", inner))
+				} else {
+					None
+				}
+			}
+			TypeAnnotationKind::Optional(t) => {
+				if let Some(inner) = self.jsify_type(&t.kind) {
+					Some(format!("{}?", inner))
+				} else {
+					None
+				}
+			}
+			_ => None,
+		}
+	}
+
+	// This helper determines what requirement and dependency to add to the struct schema based
+	// on the type annotation of the field.
+	// I.E. if a struct has a field named "foo" with a type "OtherStruct", then we want to add
+	// the field "foo" as a required  and the struct "OtherStruct" as a dependency. so the result is
+	// a tuple (required, dependency)
+	fn extract_struct_schema_dependencies(
+		&self,
+		typ: &TypeAnnotationKind,
+		field_name: &String,
+	) -> Option<(Option<String>, String)> {
+		match typ {
+			TypeAnnotationKind::UserDefined(udt) => Some((Some(field_name.clone()), udt.root.name.clone())),
+			TypeAnnotationKind::Array(t) => self.extract_struct_schema_dependencies(&t.kind, field_name),
+			TypeAnnotationKind::Optional(t) => {
+				if let Some(deps) = self.extract_struct_schema_dependencies(&t.kind, field_name) {
+					// We never want to add an optional to the required block
+					Some((None, deps.1))
+				} else {
+					None
+				}
+			}
+			_ => None,
+		}
+	}
+
+	fn jsify_struct_field_to_json_schema_type(&self, typ: &TypeAnnotationKind, field_name: &String) -> String {
+		match typ {
+			TypeAnnotationKind::Bool | TypeAnnotationKind::Number | TypeAnnotationKind::String => {
+				format!("type: \"{}\"", self.jsify_type(typ).unwrap())
+			}
+			TypeAnnotationKind::UserDefined(udt) => {
+				format!("\"$ref\": \"/{}\"", udt.root.name)
+			}
+			TypeAnnotationKind::Array(t) => {
+				format!(
+					"type: \"array\", items: {{ {} }}",
+					self.jsify_struct_field_to_json_schema_type(&t.kind, field_name)
+				)
+			}
+			TypeAnnotationKind::Optional(t) => self.jsify_struct_field_to_json_schema_type(&t.kind, field_name),
```

Comment:
Where in the schema do we specify this is optional?

Reply:
Everything in the schema is optional, unless required. So I add a required entry for everything that is not optional. If the snapshots had the generated schema file it would be clearer 😅 I’ll get that added